kong/dynamic_hook/init.lua

```diff
@@ -0,0 +1,123 @@
+local warp_function_gen = require("kong.dynamic_hook.wrap_function_gen")
+
+local _M                = {
+  TYPE = {
+    BEFORE = 1,
+    AFTER  = 2,
+    BEFORE_MUT = 3,
+    AFTER_MUT  = 4,
+  },
+}
+
+local pcall             = pcall
+
+local non_function_hooks = {
+--[[
+  [group_name] = {
+    [hook_name] = <function>,
+    ...
+  },
+  ...
+--]]
+}
+
+local always_enabled_groups = {}
+
+local wrap_functions = {
+  [0] = warp_function_gen.generate_wrap_function(0),
+  [1] = warp_function_gen.generate_wrap_function(1),
+  [2] = warp_function_gen.generate_wrap_function(2),
+  [3] = warp_function_gen.generate_wrap_function(3),
+  [4] = warp_function_gen.generate_wrap_function(4),
+  [5] = warp_function_gen.generate_wrap_function(5),
+  [6] = warp_function_gen.generate_wrap_function(6),
+  [7] = warp_function_gen.generate_wrap_function(7),
+  [8] = warp_function_gen.generate_wrap_function(8),
+  ["varargs"] = warp_function_gen.generate_wrap_function("varargs"),
```

Comment:
Could we define `varargs` as `-1`?

Reply:
This code path is not hot, so readability is more important.